CodeStarterCamp_Week3/main.swift

```diff
@@ -35,32 +35,34 @@ class Person {
     }
     
     func buy(coffee: Coffee, from coffeeShop: CoffeeShop) {
-        if Coffee.allCases.contains(coffee) {
-            let hasEnoughMoney = self.money >= coffee.price
-            guard hasEnoughMoney else {
-                print("잔액이 \(coffee.price-self.money)원만큼 부족합니다.")
-                return
-            }
-            self.money -= coffee.price
-            print("결제가 완료되었습니다. 잔액은 \(self.money)원입니다.")
-            coffeeShop.takeOrder(coffee: coffee, from: self)
-        }
-        else {
+        guard coffeeShop.menu.contains(coffee) else {
             print("죄송합니다. 주문하신 \(coffee)는 판매하지 않습니다.")
+            return
+        }
```

Comment:
말씀해주신대로 `CoffeeShop`의 `checkCoffee`가 판매중인 커피인지 확인 후 `Bool` 값을 반환할테니
```suggestion
        guard coffeeShop.checkCoffee(coffee) else { return }
```
이런 형태가 되겠네요 🙂 
`print("죄송합니다. 주문하신 \(coffee)는 판매하지 않습니다.")` 와 같은 커피 판매에 대한 안내 문구도 커피숍의 메서드로 옮겨가면 더 좋을 것 같구요!